### DISCUSSION TITLE
Game crashing teleporting

### DISCUSSION THREAD

**Discussion by Nianham1** [runelite/runelite#18390] [Q&A]
Since ~ 10/3/2024, just about every time I finish an Araxxor trip and use my home teleport tab, the game crashes entirely and is unresponsive. I have to kill runelite and restart for it to work.

**Comment by Nianham1** [comment]
Update: This is not limited to teleport tabs or Araxxor, but seemingly occurs randomly any time I teleport to my house through any means.

**Comment by Nianham1** [comment]
Update: Checking client logs, this looks to be caused by the GPU plugin?

2024-10-05 11:56:37 EDT [Client] ERROR injected-client - Client error
java.lang.NullPointerException: null
	at net.runelite.client.plugins.gpu.GpuPlugin.swapScene(GpuPlugin.java:1587)
	at client.pw(client.java:1837)
	at client.kl(client.java:23812)
	at client.ij(client.java:3194)
	at client.bj(client.java:1138)
	at ba.aw(ba.java:409)
	at ba.gr(ba.java)
	at ba.run(ba.java:11873)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by Nianham1** [comment]
new crash with different plugin

2024-10-05 17:04:27 EDT [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 104
	at com.stopmisclickingtiles.StopMisclickingTilesPlugin.onMenuOptionClicked(StopMisclickingTilesPlugin.java:199)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:70)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:223)
	at net.runelite.client.callback.Hooks.post(Hooks.java:202)
	at qt.mo(qt.java:48922)
	at vl.io(vl.java:99)
	at vl.yd(vl.java)
	at client.wc(client.java:57906)
	at client.gn(client.java:24562)
	at client.ij(client.java:3583)
	at client.bj(client.java:1138)
	at ba.aw(ba.java:409)
	at ba.gr(ba.java)
	at ba.run(ba.java:11873)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by cdfisher** [comment]
You're probably just running out of heap space. Can you post your full log instead of just snippets so we can take a look at it? Just copying small sections tends to remove important context.

**Comment by kylestev** [comment]
I have also been encountering this issue since yesterday. Prior to yesterday, I last used RuneLite on 9/30 and my client log file is almost empty. I am experiencing similar symptoms where my RuneLite client using a Jagex Account launched from the Jagex Launcher on macOS is unable to stay logged in for long periods of time.

Issues experienced come in one of two flavors:
1. My client will freeze and take me to the login screen without safely logging out of my account
1. My client will freeze and show `"Connection lost.\nPlease wait - attempting to reestablish"` in a black box in the upper left corner of the screen momentarily before either going to the login screen, or continuing gameplay

I've tried to narrow down plugins that may be causing this issue.

today and yesterday's logs (lightly modified to redact RSN and Discord handles) https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7


snippets to look at:
- `client_2024-10-04.0.log`
    - `2024-10-05 12:53:49 PDT [Client] ERROR injected-client - Client error: 90,79,114,6,1637,3492,-22,53,52,28,-121,0, java.lang.ArrayIndexOutOfBoundsException: Index -5579 out of bounds for length 889` ([L28-29](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client_2024-10-04-0-log-L28-L29))
    - `2024-10-05 12:59:20 PDT [Client] ERROR injected-client - Client error: 64,45,129,10,1637,3492,-1,-1,94,-1,-1,-110,72,0,0,0, java.lang.ArrayIndexOutOfBoundsException: null` ([L39-39](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client_2024-10-04-0-log-L38-L39))
    - `2024-10-05 13:01:36 PDT [Client] ERROR injected-client - Client error: 108,94,119,1362,1637,3492,-83,88,-1,0,5,-22,-1,-1,66,30,0,0,0,0,19,19,-1,-1,124,-1,-1,103,-2,0,0,0,0,91,53,52,-39,-108,0,1,0,-96,0,18,-64,109,107,17,-104,27,34,-4,52,65,-110,-42, java.lang.IllegalStateException: null` ([L40-41](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client_2024-10-04-0-log-L40-L41))
- `client.log` (10/5/24, last updated about 15 minutes ago as of writing)
    - Frequency
        -  `ArrayIndexOutOfBoundsException` x18
        - `InaccessibleObjectException` x2
        - `BindException` x4 (probably because i had two clients open?)
        - `NullPointerException` x7
        - `IllegalComponentStateException` x6 (all seem to be related to the Picture-in-Picture plugin which I have since removed)
        - `OutOfMemoryError` x1 (when i looked at the Activity Monitor on macOS, the usage of the client was less than 2.5gb. Sometimes runelite gets up to 16gb on my mac if i leave it open for a week)
    - `2024-10-05 15:16:23 PDT [Client] ERROR injected-client - Client error: 56,45,129,8,2723,3502,15,40,20,20,0,0,34,-1, java.lang.ArrayIndexOutOfBoundsException: Index 8959 out of bounds for length 889` ([L1329](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client-log-L1329))
    - `2024-10-05 15:02:07 PDT [Client] ERROR injected-client - Client error: 29,45,129,6,2729,3502,-1,-1,42,-1,-1,42, java.lang.ArrayIndexOutOfBoundsException: Index 11007 out of bounds for length 889` ([L1166-1167](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client-log-L1166))
    - `2024-10-05 15:00:03 PDT [Client] ERROR injected-client - Client error java.lang.NullPointerException: Cannot invoke "jh.nl(int[], int, int, int, int)" because "<parameter1>.cp" is null` ([L1061-1062](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client-log-L1061-L1062))
    - `2024-10-05 15:00:02 PDT [Client] ERROR injected-client - Client error: 57,95,27,5,2729,3502,-128,3,1,9,99, java.lang.ArrayIndexOutOfBoundsException: null` ([L1059-1060](https://gist.github.com/kylestev/ba44f1740f8acf4020115e3490ac30b7#file-client-log-L1059-L1060))

**Reply by cdfisher** [reply]
> OutOfMemoryError x1 (when i looked at the Activity Monitor on macOS, the usage of the client was less than 2.5gb. Sometimes runelite gets up to 16gb on my mac if i leave it open for a week)

The OOM is you running out of JVM heap space. Disable some hub plugins. The high memory usage is a GPU driver memory leak. In any case, we recommend closing your client at least once a day to avoid issues and ensure you receive updates.

Are you using Sequoia? Both of your main points sound like connection issues that a lot of users on Sequoia have reported.

**Reply by kylestev** [reply]
I am using Sequoia. Updated yesterday to the latest public patch: macOS 15.0.1 (24A348)

FWIW I also have Little Snitch 6.1.1 but RuneLite is generally allow listed for network requests, same with Jagex Launcher (with the exception of some ad/tracking services).

After reading some public forums it looks like there might be a few options - none of which I am enthused with as they either involve disabling SIP or reinstalling the OS. Is there anything I can do to enable more logging on my side so I can submit feedback to Apple?

**Reply by cdfisher** [reply]
I don't think we have a known-good fix for the Sequoia issues other than that reinstalling has seemed to help some people. I don't think any RuneLite related logging would be helpful to Apple since this seems to be a broader issue.
